datafusion/optimizer/src/analyzer/mod.rs

```diff
@@ -155,7 +155,7 @@ impl Analyzer {
 
 /// Do necessary check and fail the invalid plan
 fn check_plan(plan: &LogicalPlan) -> Result<()> {
-    plan.apply(&mut |plan: &LogicalPlan| {
+    plan.apply_with_subqueries(&mut |plan: &LogicalPlan| {
```

Comment:
It is interesting that this didn't change any plans / tests.

Reply:
Basically I just moved the code of `LogicalPlan:: apply()` (that overrided `TreeNode:: apply()`) to `LogicalPlan:: apply_with_subqueries()` and changed the `LogicalPlan:: apply()` calls to `LogicalPlan:: apply_with_subqueries()`. I updated the PR description to be clear.